pandas/core/case_when.py

```diff
@@ -0,0 +1,217 @@
+from __future__ import annotations
+
+from itertools import zip_longest
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.cast import (
+    construct_1d_arraylike_from_scalar,
+    find_common_type,
+    infer_dtype_from,
+)
+from pandas.core.dtypes.common import is_scalar
+from pandas.core.dtypes.generic import ABCNDFrame
+
+from pandas.core.construction import array as pd_array
+
+if TYPE_CHECKING:
+    from pandas._typing import Series
+
+
+def case_when(
+    *args: tuple[tuple],
+    default=lib.no_default,
+    level: int | None = None,
+) -> Series:
+    """
+    Replace values where the conditions are True.
+    It is a convenience wrapper around `pd.Series.mask`,
+    and is handy for multiple conditions and replacements.
+
+    Parameters
+    ----------
+    *args : array-like, scalar
+        Variable argument of tuples of conditions and expected replacements.
+        Takes the form:  `(condition0, replacement0)`,
+            `(condition1, replacement1)`, ... .
+        `condition` should be a 1-D boolean array.
+        When multiple boolean conditions are satisfied,
+        the first replacement is used.
+
+    default : scalar, array-like, default None
+        If provided, it is the replacement value to use
+        if all conditions evaluate to False.
+        If not specified, entries will be filled with the
+        corresponding NULL value (``np.nan`` for numpy dtypes,
+        ``pd.NA`` for extension dtypes).
+
+    level : int, default None
+        Alignment level if needed.
```

Comment:
When used with a single condition, you can use mask. When used with multiple conditions, it seems to me to be rare that `level` will be useful. I would suggest not exposing this if there isn't a compelling use case known.